wdl/tools/xenocp.wdl

```diff
@@ -0,0 +1,428 @@
+## # XenoCP
+##
+## This WDL tool wraps [XenoCP](https://github.com/stjude/XenoCP).
+## XenoCP is cleans contamination reads from a BAM file.
+
+version 1.0
+
+task get_chroms {
+    input {
+        Int ncpu = 1
+        File input_bam
+        Int memory_gb = 1
+        Int? disk_size_gb
+        Int max_retries = 1
+        Boolean detect_nproc = false
+    }
+
+    String parsed_detect_nproc = if detect_nproc then "true" else ""
+    Float input_bam_size = size(input_bam, "GiB")
+    Int disk_size = select_first([disk_size_gb, ceil(input_bam_size + 2)])
+
+    command <<<
+        set -euo pipefail
+
+        n_cores=~{ncpu}
+        if [ -n ~{parsed_detect_nproc} ]
+        then
+            n_cores=$(nproc)
+        fi
+
+        samtools view -H ~{input_bam} -@ ${n_cores}| grep "@SQ" | cut -f 2  | cut -f 2 -d':' > chroms.txt
```

Comment:
We don't need to know number of CPUs or cores for this step, right?  It's just a samtools view -H.

Reply:
We don't need to, but it can multithread, so I've enabled it.